textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/writer/LuceneDataWriter.java

```diff
@@ -0,0 +1,102 @@
+package edu.uci.ics.textdb.storage.writer;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.FSDirectory;
+
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.FieldType;
+import edu.uci.ics.textdb.api.common.IField;
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.storage.IDataWriter;
+import edu.uci.ics.textdb.common.exception.StorageException;
+import edu.uci.ics.textdb.common.utils.Utils;
+import edu.uci.ics.textdb.storage.LuceneDataStore;
+
+public class LuceneDataWriter implements IDataWriter{
+    
+    private String indexDir;
+
+    public LuceneDataWriter(String indexDir) {
+        this.indexDir = indexDir;
+    }
+    
+    @Override
+    public void clearData() throws StorageException{
+        IndexWriter indexWriter = null;
+        try {
+            Directory directory = FSDirectory.open(Paths
+                    .get(indexDir));
+            Analyzer analyzer = new StandardAnalyzer();
+            IndexWriterConfig conf = new IndexWriterConfig(analyzer);
+            indexWriter = new IndexWriter(directory, conf);
+            indexWriter.deleteAll();
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new StorageException(e.getMessage(), e);
+        } finally {
+            if (indexWriter != null) {
+                try {
+                    indexWriter.close();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                    throw new StorageException(e.getMessage(), e);
+                }
+            }
+        }
+        
+    }
+    
+    @Override
+    public void writeData(List<Attribute> schema, List<ITuple> tuples) throws StorageException {
+        IndexWriter indexWriter = null;
+        try {
+            Directory directory = FSDirectory.open(Paths
+                    .get(indexDir));
+            Analyzer analyzer = new StandardAnalyzer();
+            IndexWriterConfig conf = new IndexWriterConfig(analyzer);
+            indexWriter = new IndexWriter(directory, conf);
+            
+            for (ITuple sampleTuple : tuples) {
+
+                Document document = getDocument(schema, sampleTuple);
+                indexWriter.addDocument(document);
+            }
+            LuceneDataStore.incrementNumDocuments(tuples.size());

```

Comment:
Given this, I think the previous discussion about the `IDataStore` is useful, or at least we should have a `LuceneDataStore` class instead of using it statically. 

Furthermore, the `DataStore` should have the API of `getDirectory`, `getSchema`, `getNumberDoc` `incNumberDoc` ... etc, and the `IDataWriter` should use `DataStore` as a parameter. `IDataWriter` will get the directory and schema from the `DataStore`. Then the API of `writeData` should be simplified as `writeData(List<ITuple> tuples)`.